src/scripts/main.js

```diff
@@ -1,3 +1,276 @@
 'use strict';
 
 // write your code here
+
+const button = document.querySelector('.button');
+const line = document.querySelectorAll('tr'); // [4]
+const eachTd = document.querySelectorAll('td');
+const messageLose = document.body.querySelector('.message-lose');
+const messageWin = document.body.querySelector('.message-win');
+
+button.addEventListener('click', () => {
+  button.textContent = 'Restart';
+  button.classList.remove('start');
+  button.classList.add('restart');
+  button.classList.add('restart');
+  messageWin.classList.add('hidden');
+  messageLose.classList.add('hidden');
+  resetAll('all');
+  gameStart();
+});
+
+function randomStand(freePosition) {
+  for (const element of eachTd) {
+    if (element.textContent === '') {
+      freePosition.push(element);
+    }
+  }
+
+  if (freePosition.length === 0) {
+    gameFinish();
+  }
+
+  let randomCount = 1;
+
+  do {
+    randomCount = Math.round(Math.random() * (4 - 2) + 2);
+  } while (randomCount % 2 === 1);
+
+  const randomElement = Math.round(
+    Math.random() * (freePosition.length - 1) + 0
+  );
+
+  changeElement(freePosition[randomElement], randomCount);
+}
+// создание рандомных значений
+
+function gameStart() {
+  const message = document.body.querySelector('.message-start');
+
+  message.classList.add('hidden');
+
+  const freePosition = [];
+
+  randomStand(freePosition);
+
+  document.addEventListener('keydown', pushElement);
+} // основа для запуска
+
+function gameFinish() {
+  let resultScore = 0;
+
+  for (const item of eachTd) {
+    if (item.textContent > 0) {
+      resultScore += parseInt(item.textContent, 10);
+    }
+  }
+
+  const score = document.body.querySelector('.game-score');
+
+  if (parseInt(score.textContent, 10) > resultScore) {
+    messageWin.classList.add('hidden');
+    messageLose.classList.remove('hidden');
+  } else if (parseInt(score.textContent, 10) < resultScore) {
+    score.textContent = resultScore;
+    messageLose.classList.add('hidden');
+    messageWin.classList.remove('hidden');
+  }
```

Comment:
avoid such nesting `else if` conditions
This code is hard to read and maintain

Reply:
Ok.